### PR TITLE
amarel cfg

### PR DESCRIPTION
**Pull request by andre-merzky**
as discussed: use `exclusive` and `exact` options on amatel.

@mtitov : Maybe we can change the check in the `srun` LM to something like

```py
if submitted_with_exclusive:
    self._exact == True
```

?

### CODE REVIEW

**Comment by mtitov** [comment]
> @mtitov : Maybe we can change the check in the `srun` LM to something like
> 
> ```python
> if submitted_with_exclusive:
>     self._exact == True
> ```
> ?

yeah, that works

**Review by mtitov — APPROVED** [review]
LGTM!

**Comment by mtitov** [comment]
@andre-merzky should we try to automate this process by checking env variable`SBATCH_EXCLUSIVE` (https://slurm.schedmd.com/sbatch.html#OPT_SBATCH_EXCLUSIVE)? if it is set, then to use `exact`

**Comment by andre-merzky** [comment]
> @andre-merzky ~should we try to automate this process by checking env variable`SBATCH_EXCLUSIVE` (https://slurm.schedmd.com/sbatch.html#OPT_SBATCH_EXCLUSIVE)? if it is set, then to use `exact`~
> 
> **edit:** it is an input env variable.. looks like not applicable for us

I added code to lookup the `system_architecture` flags.  How does that look to you?

File: src/radical/pilot/agent/resource_manager/base.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ class RMInfo(ru.TypedDict):
             'mem_per_node'     : int,           # memory per node (MB)
 
             'details'          : {None: None},  # dict of launch method info
+            'sys_arch'         : {str: None},   # system architecture flags
```
Comment (mtitov) [inline src/radical/pilot/agent/resource_manager/base.py:64]: let's use `details` for this?
```python
'details': {'exact': bool(self._rcfg.get('system_architecture', {}).get('exclusive'))}
```

all other information that we have inside `system_architecture` is used in other `RMInfo` parameters, thus I would go for to extract it and not to preserve the original state, especially `details` is already used for additional data for LM

**Pull request merged**